boefjes/tools/upgrade_v1_16_0.py

```diff
@@ -0,0 +1,168 @@
+    meaning that it can be rerun until there are no, or only expected, exceptions.
+    """
+
+    organisations = create_organisation_storage(sessionmaker(bind=get_engine())()).get_all()
+    valid_time = datetime.now(timezone.utc)
+
+    bytes_client = get_bytes_client()
+    bytes_client.login()
+
+    total_failed = 0
+    total_processed = 0
+    logger.info("Processing %s organisations in total", len(organisations))
+
+    for organisation_id in organisations:
+        connector = get_octopoes_api_connector(organisation_id)
+        logger.info("Processing organisation [organization_id=%s]", organisation_id)
+
+        failed, processed = migrate_org(bytes_client, connector, organisation_id, valid_time)
+        total_failed += failed
+        total_processed += processed
+
+        logger.info("Processed organisation [total_processed=%s, total_failed=%s]", processed, failed)
+
+    logger.info("Finished migration [total_processed=%s, total_failed=%s]", total_processed, total_failed)
+
+
+def migrate_org(bytes_client, connector, organisation_id, valid_time) -> tuple[int, int]:
+    """
+    For each organisation, we paginate through the origin API, find the matching normalizer meta in Bytes,
+    and set the source_method to the boefje id. Then update the origin, i.e. save it and delete the old one.
+    """
+
+    failed = 0
+    processed = 0
+
+    offset = 0
+    while True:
+        origins = connector.list_origins(valid_time, offset=offset, limit=200)
+        logger.info("Processing %s origins", len(origins))
+
+        for origin in origins:
+            if origin.source_method is not None or origin.origin_type == OriginType.INFERENCE:
+                continue
+
+            try:
+                normalizer_meta = bytes_client.get_normalizer_meta(origin.task_id)
+                origin.source_method = normalizer_meta.raw_data.boefje_meta.boefje.id
+                update_origin(connector, origin, valid_time)
+            except HTTPStatusError as error:
+                # We expect to find Declaration/Affirmations without a normalizer meta
+                if error.response.status_code == 404 and origin.method != "manual":
+                    logger.warning(
+                        "Could not find normalizer_meta [task_id=%s, method=%s, origin_type=%s]",
+                        origin.task_id,
+                        origin.method,
+                        origin.origin_type,
+                    )
+                elif error.response.status_code == 404:
+                    logger.info(
+                        "Could not find normalizer_meta [task_id=%s, method=%s, origin_type=%s]",
+                        origin.task_id,
+                        origin.method,
+                        origin.origin_type,
+                    )
+                else:
+                    logger.exception(
+                        "Could not update origin [task_id=%s, method=%s, origin_type=%s]",
```

Comment:
This seems to say that the `update_origin` fails, but it might also be that `get_normalizer_meta` returns something else than 404. I think it would be cleaner to have a try/except around the `get_normalizer_meta` and another try/except around `update_origin`.